### PR TITLE
Bug 2044140: Updated the error message to include the suggestion

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

**Review by wking — APPROVED** [review]
/lgtm
/hold

Fix looks good to me.  Holding in case folks want to do any pre-merge testing, but I'm fine with this merging whenever.

**Comment by LalatenduMohanty** [comment]
Tested the PR and the output looks like below
```
$ ./oc adm upgrade --to-image registry.ci.openshift.org/ocp/release@sha256:f1b711b1f476d021d0bdc2b88cd28d5b92e4459b49c787db2129c6c1aaf26652
error: cannot refresh available updates:
  Reason: NoChannel
  Message: The update channel has not been configured.

To continue with upgrade you must pass --allow-explicit-upgrade
```

File: pkg/cli/admin/upgrade/upgrade.go [hunk 1]
```diff
@@ -305,12 +305,29 @@ func (o *Options) Run() error {
...
+				if o.ToImage != "" {
+					return errors.New("no recommended updates, specify --allow-explicit-upgrade to continue with the update" +
+						" or wait for new updates to be available")
+				}
 				return errors.New("no recommended updates, specify --to-image or wait for new updates to be available")
 			case len(possibleUpgradeTargets) > 0:
+				if o.ToImage != "" {
+					return fmt.Errorf("the update is not one of the possible targets: %s. Specify --allow-explicit-upgrade to continue with the update",
+						strings.Join(possibleUpgradeTargets, ", "))
+
```
Comment (wking) [inline pkg/cli/admin/upgrade/upgrade.go:329]: nit: empty line

**Review by wking — APPROVED** [review]

**Comment by wking** [comment]
[volumeLimits][1], [`KubeJobFailed`, and console connection flakes][2] issues are all orthogonal.  I'm probably not authorized, but trying anyway:

/override ci/prow/e2e-aws-serial
/override ci/prow/e2e-aws-upgrade

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_oc/1041/pull-ci-openshift-oc-master-e2e-aws-serial/1488719657329233920
[2]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_oc/1041/pull-ci-openshift-oc-master-e2e-aws-upgrade/1488719657366982656

**Pull request merged**